x/deployment/client/cli/query.go

```diff
@@ -21,67 +20,105 @@ func GetQueryCmd(key string, cdc *codec.Codec) *cobra.Command {
 		RunE:                       client.ValidateCmd,
 	}
 
-	cmd.AddCommand(flags.GetCommands(
-		cmdDeployments(key, cdc),
-		cmdDeployment(key, cdc),
-		getGroupCmd(key, cdc),
-	)...)
+	cmd.AddCommand(
+		cmdDeployments(),
+		cmdDeployment(),
+		getGroupCmd(),
+	)
 
 	return cmd
 }
 
-func cmdDeployments(key string, cdc *codec.Codec) *cobra.Command {
+func cmdDeployments() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "list",
 		Short: "Query for all deployments",
 		Args:  cobra.ExactArgs(0),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			ctx := context.NewCLIContext().WithCodec(cdc)
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			clientCtx, err := client.ReadQueryCommandFlags(clientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
 
-			dfilters, err := DepFiltersFromFlags(cmd.Flags())
+			queryClient := types.NewQueryClient(clientCtx)
+
+			dfilters, state, err := DepFiltersFromFlags(cmd.Flags())
 			if err != nil {
 				return err
 			}
 
-			obj, err := query.NewClient(ctx, key).Deployments(dfilters)
+			// checking state flag
+			stateVal, ok := types.Deployment_State_value[state]
+
+			if (!ok && (state != "")) || state == "invalid" {
```

Comment:
use `stateVal` for the second comparison.

Reply:
We defined state with 0 value as InvalidState in types. So, when user enters `invalid` as state filter value, then it will considered as a State and will return all values. To solve this, I am checking entered value not to be `invalid`.

Reply:
yeah, can you check `stateVal == types.DeploymentState_Invalid` (or whatever the constant's name is) instead of the string comparison?  It's easier to keep track of that way.

Reply:
No,`stateval` will be equal to `types.DeploymentState_Invalid` when state filter is not applied. So we cannot use stateval to compare here.

Reply:
if `state == "invalid"`, won't `stateVal == types.DeploymentState_Invalid`?

Reply:
Even if `state==" "`, `stateVal == types.DeploymentState_Invalid` because by default, stateVal will be 0

Reply:
That would be handled by `!ok && state != ""`. Also, people shouldn't be typing `"invalid"` ever - they should leave it as blank to get all results.

Reply:
If people type invalid then ok will be true. That's the reason I am handling it to just avoid such cases. If not needed, I can remove it but you eill get all results if state filter value entered is `invalid`.

Reply:
Right.  I was asking for the second part to use the constant.  You keep saying that it won't work for whatever reason.

Forget it.  Maybe I'm missing something.  I'll change it myself if not.

Reply:
Sorry, I'm an idiot.  You're right.